src/Microsoft.JSInterop/JavaScriptRuntime/src/Microsoft.JSInterop.ts

```diff
@@ -0,0 +1,234 @@
+// This is a single-file self-contained module to avoid the need for a Webpack build
+
+module DotNet {
+  (window as any).DotNet = DotNet; // Ensure reachable from anywhere
+
+  export type JsonReviver = ((key: any, value: any) => any);
+  const jsonRevivers: JsonReviver[] = [];
+
+  const pendingAsyncCalls: { [id: number]: PendingAsyncCall<any> } = {};
+  const cachedJSFunctions: { [identifier: string]: Function } = {};
+  let nextAsyncCallId = 1; // Start at 1 because zero signals "no response needed"
+
+  let dotNetDispatcher: DotNetCallDispatcher | null = null;
+
+  /**
+   * Sets the specified .NET call dispatcher as the current instance so that it will be used
+   * for future invocations.
+   *
+   * @param dispatcher An object that can dispatch calls from JavaScript to a .NET runtime.
+   */
+  export function attachDispatcher(dispatcher: DotNetCallDispatcher) {
+    dotNetDispatcher = dispatcher;
+  }
+
+  /**
+   * Adds a JSON reviver callback that will be used when parsing arguments received from .NET.
+   * @param reviver The reviver to add.
+   */
+  export function attachReviver(reviver: JsonReviver) {
+    jsonRevivers.push(reviver);
+  }
+
+  /**
+   * Invokes the specified .NET public method synchronously. Not all hosting scenarios support
+   * synchronous invocation, so if possible use invokeMethodAsync instead.
+   *
+   * @param assemblyName The name of the .NET assembly containing the method.
```

Comment:
Do you think it's worth specifying in the docs that its the short assembly name (without key and version)? If you're coming from a .NET background you might assume that it's the full name

Reply:
Good point. Done.